php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1011 @@
+      f.functionDeclaration(
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        BLOCK()));
+  }
+  
+  public ParameterListTree PARAMETER_LIST() {
+    return b.<ParameterListTree>nonterminal(PHPLexicalGrammar.PARAMETER_LIST).is(
+      f.parameterList(
+        b.token(PHPPunctuator.LPARENTHESIS),
+        b.optional(
+          f.newTuple94(
+            PARAMETER(), 
+            b.zeroOrMore(
+              f.newTuple93(
+                b.token(PHPPunctuator.COMMA), 
+                PARAMETER())))),
+        b.token(PHPPunctuator.RPARENTHESIS)));
+  }
+  
+  public ParameterTree PARAMETER() {
+    return b.<ParameterTree>nonterminal(PHPLexicalGrammar.PARAMETER).is(
+      f.parameter(
+        b.optional(
+          b.firstOf(
+            b.token(PHPKeyword.ARRAY),
+            b.token(PHPKeyword.CALLABLE),
+            NAMESPACE_NAME())),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        b.optional(b.token(PHPPunctuator.ELIPSIS)),
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.optional(
+          f.newTuple92(
+            b.token(PHPPunctuator.EQU), 
+            STATIC_SCALAR())))
+      );      
+  }
+
+  /**
+   * [ END ] Declaration
+   */
+
+  /**
+   * [ START ] Statement
+   */
+
+  public StatementTree TOP_STATEMENT() {
+    return b.<StatementTree>nonterminal(PHPLexicalGrammar.TOP_STATEMENT)
+        .is(b.firstOf(
+//            CLASS_DECLARATION(),
+            FUNCTION_DECLARATION(),
+//            INTERFACE_DECLARATION(),
+            NAMESPACE_STATEMENT(),
+            USE_STATEMENT(),
+//            CONSTANT_DECLARATION(),
+            STATEMENT()
+        ));
+  }
+
+  public NamespaceStatementTree NAMESPACE_STATEMENT() {
+    return b.<NamespaceStatementTree>nonterminal(PHPLexicalGrammar.NAMESPACE_STATEMENT)
+        .is(b.firstOf(
+           f.namespaceStatement(
+               b.token(PHPKeyword.NAMESPACE),
+               NAMESPACE_NAME(),
+               EOS()
+           ),
+           f.blockNamespaceStatement(
+               b.token(PHPKeyword.NAMESPACE),
+               b.optional(NAMESPACE_NAME()),
+               b.token(LCURLYBRACE),
+               b.zeroOrMore(TOP_STATEMENT()),
+               b.token(RCURLYBRACE)
+           )
+        ));
+  }
+
+  public UseStatementTree USE_STATEMENT() {
+    return b.<UseStatementTree>nonterminal(PHPLexicalGrammar.USE_STATEMENT).is(
+        f.useStatement(
+            b.token(PHPKeyword.USE),
+            b.optional(
+                b.firstOf(
+                    b.token(PHPKeyword.CONST),
+                    b.token(PHPKeyword.FUNCTION))),
+            USE_DECLARATION(),
+            b.zeroOrMore(f.newTuple90(b.token(PHPPunctuator.COMMA), USE_DECLARATION())),
+            EOS())
+    );
+  }
+
+  public StatementTree STATEMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)